### PR TITLE
Add Note Command

### LINKED ISSUE #2: User Story - Add Note Command
As a computing student, I want to add notes about my internship applications so that I can jot down important information about the company / application process.

Refer to UC8 Use case in developer guide. 

Follow [Add remark tutorial](https://nus-cs2103-ay2324s1.github.io/tp/tutorials/AddRemark.html)

### PR DESCRIPTION
**Pull request by AprupKale**
Fixes #2 

Add the note command, and update the logic and model accordingly

### CODE REVIEW

**Comment by AprupKale** [comment]
Yet to integrate with Storage and finalize the other classes

Not ready to merge!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -152,9 +152,33 @@ Updates an internship application entry with the specified status.
...
+
+**Format**: `note INDEX [i/NOTE]`
+
+**Parameters**:
+* `INDEX`: Index of internship application entry to be updated
+    * `INDEX` must be a positive integer, such as, `1`, `2`, `3`, etc.
+* `i/NOTE`: Note to be added to the internship application
+
+**Examples**:
+* `note 1 i/John Street is the leading market maker in the APAC region` adds the note 'John Street is the leading market maker in the APAC regio' to first internship application entry (in the list)
```
Comment (tim-pipi) [inline docs/UserGuide.md:171]:
```suggestion
* `note 1 i/John Street is the leading market maker in the APAC region` adds the note 'John Street is the leading market maker in the APAC region' to the first internship application entry (in the list)
```

Typo here!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -209,11 +233,12 @@ If your changes to the data file makes its format invalid, LetsGetHired will dis
...
-| **Delete** | `delete INDEX`<br> e.g., `delete 1`                                                                                     |
-| **List**   | `list`                                                                                                                  |
-| **Help**   | `help`                                                                                                                  |
+| Action     | Format, Examples                                                                                                       |
+|------------|------------------------------------------------------------------------------------------------------------------------|
+| **Add**    | `add c/CYCLE r/ROLE n/COMPANY_NAME [s/STATUS]` <br> e.g., `add c/Summer r/Full Stack Developer n/John street s/Applied`|
+| **View**   | `view INDEX`<br> e.g., `view 3`                                                                                        |
+| **Tag**    | `tag INDEX`<br> e.g.,`tag 2 s/Applied`
+| **Note**   | `note INDEX`<br> e.g., `note 5 i/John Street is the leading market maker in the APAC region`
```
Comment (tim-pipi) [inline docs/UserGuide.md:241]: I think these lines are missing the right `|`

File: src/main/java/seedu/letsgethired/logic/parser/AddCommandParser.java [hunk 3]
```diff
@@ -10,11 +10,7 @@
 
 import seedu.letsgethired.logic.commands.AddCommand;
 import seedu.letsgethired.logic.parser.exceptions.ParseException;
-import seedu.letsgethired.model.application.Company;
-import seedu.letsgethired.model.application.Cycle;
-import seedu.letsgethired.model.application.InternApplication;
-import seedu.letsgethired.model.application.Role;
-import seedu.letsgethired.model.application.Status;
+import seedu.letsgethired.model.application.*;
```
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/logic/parser/AddCommandParser.java:13]: Should these remain specified individually (checkstyle error)?

File: src/main/java/seedu/letsgethired/logic/parser/AddCommandParser.java [hunk 4]
```diff
@@ -40,8 +36,9 @@ public AddCommand parse(String args) throws ParseException {
         Role role = ParserUtil.parseRole(argMultimap.getValue(PREFIX_ROLE).get());
         Cycle cycle = ParserUtil.parseCycle(argMultimap.getValue(PREFIX_CYCLE).get());
         Status status = ParserUtil.parseStatus(argMultimap.getValue(PREFIX_STATUS).get());
+        Note note = new Note("");
```
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/logic/parser/AddCommandParser.java:39]: Should this use `ParserUtil.parseNote` instead? Unless if you haven't implemented it yet?

File: src/main/java/seedu/letsgethired/model/util/SampleDataUtil.java
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/model/util/SampleDataUtil.java:1]: For the Sample Data, I think it's better to specify it individually instead!

File: src/main/java/seedu/letsgethired/storage/JsonAdaptedInternApplication.java [hunk 5]
```diff
@@ -84,6 +81,8 @@ public InternApplication toModelType() throws IllegalValueException {
         }
         final Status modelStatus = new Status(status);
 
-        return new InternApplication(modelCompany, modelRole, modelCycle, modelStatus);
+        final Note modelNote = new Note("");
```
Comment (tim-pipi) [inline src/main/java/seedu/letsgethired/storage/JsonAdaptedInternApplication.java:84]: I think there should be a `@JsonProperty("note")` in the constructor of this file? And also a `note` field as well?

Perhaps we could also try to follow the other fields to specify this, for example:
```suggestion
        if (note == null) {
            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Note.class.getSimpleName()));
        }
        if (!Note.isValidNote(note)) {
            throw new IllegalValueException(Note.MESSAGE_CONSTRAINTS);
        }
        final Note modelNote = new Note(note);
```

File: src/test/java/seedu/letsgethired/logic/parser/InternTrackerParserTest.java [hunk 6]
```diff
@@ -13,10 +14,7 @@
 
 import org.junit.jupiter.api.Test;
 
-import seedu.letsgethired.logic.commands.AddCommand;
-import seedu.letsgethired.logic.commands.ClearCommand;
-import seedu.letsgethired.logic.commands.DeleteCommand;
-import seedu.letsgethired.logic.commands.EditCommand;
+import seedu.letsgethired.logic.commands.*;
```
Comment (tim-pipi) [inline src/test/java/seedu/letsgethired/logic/parser/InternTrackerParserTest.java:17]: Import statement issue here too! Saw this in a few other places as well! Perhaps we could try using the IntelliJ Checkstyle Plugin to catch these!

**Review by tim-pipi — CHANGES_REQUESTED** [review]
Hey Aprup, I think there are some part of the changes that you made that could be made more standardized - following how the rest of the commands were implemented! 

Thanks for doing this!

File: src/main/java/seedu/letsgethired/logic/parser/InternTrackerParser.java [hunk 7]
```diff
@@ -76,6 +77,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case ListCommand.COMMAND_WORD:
             return new ListCommand();
 
+        case NoteCommand.COMMAND_WORD:
+            return new NoteCommandParser().parse(arguments);
```
[the remark this answers is not in the document]
Comment (AprupKale) [reply src/main/java/seedu/letsgethired/logic/parser/InternTrackerParser.java:81]: The Note command has arguments, hence the parse(arguments); similar to Edit, View and Delete

**Comment by AprupKale** [comment]
@tim-pipi Will make the required changes. As for the check style, will make fixes after adding all the functionality as the last step for the PR! Thanks.

**Comment by AprupKale** [comment]
Need to:
- [] Make Note optional for AddCommand
- [] Add test cases for more coverage for appropriate functions

**Pull request merged**